Content.Client/Administration/UI/AdminMenuWindow.xaml.cs

```diff
@@ -3,38 +3,61 @@
+        RobustXamlLoader.Load(this);
+        MasterTabContainer.SetTabTitle((int) TabIndex.Admin, Loc.GetString("admin-menu-admin-tab"));
+        MasterTabContainer.SetTabTitle((int) TabIndex.Adminbus, Loc.GetString("admin-menu-adminbus-tab"));
+        MasterTabContainer.SetTabTitle((int) TabIndex.Atmos, Loc.GetString("admin-menu-atmos-tab"));
+        MasterTabContainer.SetTabTitle((int) TabIndex.Round, Loc.GetString("admin-menu-round-tab"));
+        MasterTabContainer.SetTabTitle((int) TabIndex.Server, Loc.GetString("admin-menu-server-tab"));
+        MasterTabContainer.SetTabTitle((int) TabIndex.PanicBunker, Loc.GetString("admin-menu-panic-bunker-tab"));
+        /*
+         * TODO: Remove baby jail code once a more mature gateway process is established. This code is only being issued as a stopgap to help with potential tiding in the immediate future.
+         */
+        MasterTabContainer.SetTabTitle((int) TabIndex.BabyJail, Loc.GetString("admin-menu-baby-jail-tab"));
+        MasterTabContainer.SetTabTitle((int) TabIndex.Players, Loc.GetString("admin-menu-players-tab"));
+        MasterTabContainer.SetTabTitle((int) TabIndex.Objects, Loc.GetString("admin-menu-objects-tab"));
+        MasterTabContainer.OnTabChanged += OnTabChanged;
+    }
 
-        public AdminMenuWindow()
-        {
-            MinSize = new Vector2(650, 250);
-            Title = Loc.GetString("admin-menu-title");
-            RobustXamlLoader.Load(this);
-            MasterTabContainer.SetTabTitle(0, Loc.GetString("admin-menu-admin-tab"));
-            MasterTabContainer.SetTabTitle(1, Loc.GetString("admin-menu-adminbus-tab"));
-            MasterTabContainer.SetTabTitle(2, Loc.GetString("admin-menu-atmos-tab"));
-            MasterTabContainer.SetTabTitle(3, Loc.GetString("admin-menu-round-tab"));
-            MasterTabContainer.SetTabTitle(4, Loc.GetString("admin-menu-server-tab"));
-            MasterTabContainer.SetTabTitle(5, Loc.GetString("admin-menu-panic-bunker-tab"));
-            /*
-             * TODO: Remove baby jail code once a more mature gateway process is established. This code is only being issued as a stopgap to help with potential tiding in the immediate future.
-             */
-            MasterTabContainer.SetTabTitle(6, Loc.GetString("admin-menu-baby-jail-tab"));
-            MasterTabContainer.SetTabTitle(7, Loc.GetString("admin-menu-players-tab"));
-            MasterTabContainer.SetTabTitle(8, Loc.GetString("admin-menu-objects-tab"));
-        }
+    public event Action? OnDisposed;
 
-        protected override void Dispose(bool disposing)
-        {
-            OnDisposed?.Invoke();
-            base.Dispose(disposing);
-            OnDisposed = null;
-        }
+    private void OnTabChanged(int tabIndex)
+    {
+        TabChanged((TabIndex) tabIndex);
+    }
+
+    private void TabChanged(TabIndex tabIndex)
+    {
+        if (tabIndex == TabIndex.Objects)
+            ObjectsTabControl.RefreshObjectList();
     }
```

Comment:
If these two aren't used anywhere else, just combine them by casting to a variable inside the method.